CONTRIBUTE.md

```diff
@@ -1,18 +1,168 @@
+condition that'd [trigger](#Workload-triggers) the role as follows:
+
+Example `playbook.yml`:
+```yaml
+- hosts: localhost
+  gather_facts: no
+  tasks:
+  <existing_role_definitions>
+  - include_role:
+      name: "my-new-role"
+    when: my-new-role.condition
+```
+
+### Workload container images
+Images for workload containers must be hosted somewhere remotely accessible. You will likely reference those image URLs in your role.
+Dockerfiles to build workload images should be maintained in the [workloads](workloads/) directory as `<role_name>-Dockerfile`.
+These should be buildable by our CI system for maintaining a central public image repository.
+
+### Workload triggers
+[CRD](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) holds the definition of the resource.
+So please append your definitions to [crd](deploy/crds/bench_v1alpha1_bench_crd.yaml).
+The resources are then mentioned in a [CR](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) that operator will look for based on the CRD passed.
+The operator then triggers roles based on conditions thus [cr](deploy/crds/bench_v1alpha1_bench_cr.yaml) will influence run of [playbook](playbook.yml)
+Other vars may be defined that can modify the workload run conditions.
+
+For the sake of the example CR, please default all workloads to disabled.
+
+Example CR:
+```yaml
+apiVersion: bench.example.com/v1alpha1
+kind: Bench
+metadata:
+  name: example-bench
+spec:
+  <existing_cr_entries>
+  my-new-role:
+    # To disable, set workers to 0
+    workers: 0
+    my_key_1: my_value_1
+    my_key_2: my_value_2
+    my_dict:
+      my_key_3: my_value_3
+    when: my-new-role.condition
+```
+
+### Add workload guide
+* Keep [workload status](README.md#workloads-status) updated
+* To help users understand how the workload can be run, please add a guide similar
+to [uperf](docs/uperf.md)
+* Add the link for your workload guide to [installation guide](docs/installation.md#running-workloads)
+
+### Best practices for new workloads
+The following steps are suggested for your workload to be added:
+* The new role destroys any additional resources it created as part of role in
+case of failure or when disabled. This ensures no interference with subsequent workloads.
+* Please mention any additional cleanup required in your workload guide.
+
+## Test workload locally
+
+### The operator container image
+Any changes to the [roles](roles/) tree or to the [playbook](playbook.yml) file necessitates a new build of the operator container image.
+The container is built using the [Operator SDK](https://github.com/operator-framework/operator-sdk) and pushed to a public repository.
+The public repository could be quay in which case you'll need to:
+
+```bash
+$ operator-sdk build quay.io/<username>/benchmark-operator:testing
+$ docker push quay.io/<username>/benchmark-operator:testing
+```
+
```

Comment:
Add what `:testing` means. Just for the sake of mentioning the tags a user can define for a image.